src/hotspot/share/nmt/memTracker.hpp

```diff
@@ -51,6 +51,14 @@ class MemTracker : AllStatic {
 
  public:
 
+  static void snapshot_thread_stacks() {
+    VirtualMemoryTracker::Instance::snapshot_thread_stacks();
+  }
+
+  static bool walk_virtual_memory(VirtualMemoryWalker* walker) {
+    return VirtualMemoryTracker::Instance::walk_virtual_memory(walker);
+  }
```

Comment:
The `MemTracker` API exposes the outer and locking implementation to the rest of Hotspot. These two methods are used by us internally. I think it's better if these methods are deleted and the `VirtualMemoryTracker::Instance` methods are called directly, instead.

Reply:
Done.

Reply:
I'm not sure what you mean with using MemTracker a lot, we use it twice: Once for taking the NMT lock, and once for checking the NMT level.

The current structure of NMT is that `MemTracker` is responsible for taking locks and checking if NMT is enabled, and then calling the underlying APIs. The underlying APIs are only meant to be called directly by other NMT components, such as `MemBaseline`